sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -422,6 +441,7 @@ def analyze_property(
         is always false. For properties used in an input, can_be_plain is true,
         indicating that they can potentially be plain.
         """
+        type_string: Optional[str]
```

Comment:
Does this need to be defined at this scope level? It looks contained to the `elif` below, but I might be missing something :see_no_evil:

Reply:
That's Python scope being a little different, the variables live at the closes function level.

```python
if True:
  x = 1

if False:
  y = 2

Aprint(x) # 1
print(y) # NameError: name 'y' is not defined
```

Since the same `type_string` is used in two of the branches, we need this type annotation above, otherwise it takes the type of the first use of it (str), but the 2nd use is `Optional[str]`.

I went ahead and used 2 different name to avoid this.